src/myst.ts

```diff
@@ -62,26 +82,34 @@ export function markdownParse(text: string): Root {
   return mdast as Root;
```

Comment:
@rowanc1 I removed the change from https://github.com/executablebooks/jupyterlab-myst/pull/145 here, because I don't know how it fits into this PR, in which I've split the parsing into two routines. My hunch is the lifting logic needs to move from its current place. 

In any case, this PR is not the final API - I just want to roughly break-up the logic before tightening up the API.

Reply:
Brought the relevant changes back!